napalm_sros/sros.py

```diff
@@ -1170,23 +1180,36 @@ def _update_buff(buff):
                     return new_buff[: new_buff.rfind("\n")]
 
                 if retrieve == "running":
-                    buff_running = self._perform_cli_commands([cmd_running], True)
-                    configuration["running"] = _update_buff(buff_running)
+                    buff_running = self._perform_cli_commands(cmd_running, True)
+                    configuration["running"] = _update_buff(buff_running, skip_persistent_indices=True)
+                    if self.retrieve_bof_config:
+                        buff_bof = self._perform_cli_commands(cmd_bof_running, True)
+                        configuration["running"] += f"\n{_update_buff(buff_bof, skip_persistent_indices=False)}"
                     return configuration
                 elif retrieve == "startup":
-                    buff_running = self._perform_cli_commands([cmd_running], True)
-                    configuration["startup"] = _update_buff(buff_running)
+                    buff_running = self._perform_cli_commands(cmd_running, True)
+                    configuration["startup"] = _update_buff(buff_running, skip_persistent_indices=True)
+                    if self.retrieve_bof_config:
+                        buff_bof = self._perform_cli_commands(cmd_bof_running, True)
+                        configuration["startup"] += f"\n{_update_buff(buff_bof, skip_persistent_indices=False)}"
```

Comment:
Same remark - original code is admittedly a bit messy, but it would be better to retrieve the 'bof' config once:
```
if retrieve in ['bof','all']:
    cmd = ["admin show configuration bof | no-more"]
    buff_bof = self._perform_cli_commands(cmd, True)
    configuration["bof"] = _update_buff(buff_bof, skip_persistent_indices=False)
```

Reply:
Leaving this for now till first conversation is resolved.